### ISSUE TITLE
Release/roadmap discussion (Jan 2019)

### ISSUE THREAD

**Issue by platinumazure** [eslint/typescript-eslint-parser#597]
Not sure if an issue is the best place for this discussion, but we have good visibility here.

I'm a bit overwhelmed at the current PR backlog, so I'd like to try to clear a few things up here. I'm going to do so by proposing a release roadmap for the next 1-2 releases, and then anyone with any feedback can reply to this and we can tweak based on the feedback.

# Minor/Patch Changes

First thing I'm going to ask is-- do we even need to do any minor or patch releases with the current compatibility, given how far behind we are? Seems to me that most people using this parser would want to upgrade soon anyway, so I'm wondering if we need to do a minor or patch release in the first place. Specifically, if eslint-plugin-typescript is already starting to look at the latest typescript-estree, I'm not sure there's much value in supporting the older typescript-estree here. (@JamesHenry Thoughts on this?)

Some thoughts on specific issues that could fit in here:

* #581: If we just jump to typescript-estree@>=7, I'm not sure this is needed
* #568: Seems this could be merged now or after cutting to later typescript-estree, any thoughts on whether this is needed with current version?
* #595: There's a discussion here on whether to support `options.jsx` and `options.ecmaFeatures.jsx`. Might be better to just pick one and do a breaking change since we're close to a major release anyway.

# Major Changes

The biggest major change is to upgrade to a later version of typescript-estree, and the natural question is, which of these versions do we really need to support? I'm leaning towards just cutting one release and taking the latest (with the intention of trying to catch typescript-estree upgrades more quickly in future). @JamesHenry had suggested basically this in a comment within one of the upgrade PRs, but let me know if I misunderstood something.

If anyone has a compelling argument for releasing one of the intermediate versions, please lay it out here. Otherwise, I'll probably just merge #596, along with any other breaking changes (e.g., if #595 turns into a breaking change).

----------

Thanks everyone for your patience as we figure this out!

**Comment by j-f1** [comment]
If you want to, you could move this to a separate organization so we can coordinate releases between this, TS-ESTree, and the plugin. Additionally, you wouldn’t have to manage releases of this.

**Comment by platinumazure** [comment]
@j-f1 

> If you want to, you could move this to a separate organization so we can coordinate releases between this, TS-ESTree, and the plugin. Additionally, you wouldn’t have to manage releases of this.

Interesting you bring this up-- there are some discussions going on tangentially related to that idea. Not sure what I can say publicly yet but I'll be interested to see what we can do here.

**Comment by JamesHenry** [comment]
@platinumazure Really happy to be back supporting you on this!

I used to release this library fairly frequently, and I think I would like to return to doing that where it makes sense. It makes sense for ESLint core to major version changes very carefully and spaced apart, but for ancillary tooling like this I think it makes sense to "move fast and _fix_ things" 😄 

We have made huge improvements in `typescript-estree` (@armano2 has been fantastic) and as well as getting those in, I want to make sure anything which @uniqueiniquity and the rest of TS Team discover/raise can be swiftly addressed and released.

Unless there are any objections I would like to merge https://github.com/eslint/typescript-eslint-parser/pull/596 to set us up for the next major release and kick on from there!
